app/src/main/java/com/quran/labs/androidquran/service/AudioService.java

```diff
@@ -414,15 +414,8 @@ private void handleIntent(Intent intent) {
         audioQueue = new AudioQueue(quranInfo, audioRequest,
             new AudioPlaybackInfo(start, 1, 1, basmallah));
         Crashlytics.log("audio request has changed...");
-
-        if (player != null) {
-          player.stop();
-        }
-        state = State.Stopped;
-        Crashlytics.log("stop if playing...");
```

Comment:
stopping is sometimes the right thing to do - but this is where we should check. you have the old `audioRequest` (before we override it in line 409). we can check for example if the current verse is within the range of playback (i.e. between `oldAudioRequest.getStart()` and `oldAudioRequest.getEnd()`), then don't stop. otherwise stop, and in either case, we'll replace the `audioRequest`.

Reply:
Thanks for the reply @ahmedre I will work on the changes as soon as possible 
Ma'Assalam